mobile/src/main/java/ch/epfl/sdp/mobile/ui/features/profile/ProfileStateScreen.kt

```diff
@@ -0,0 +1,174 @@
+package ch.epfl.sdp.mobile.ui.features.profile
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.CheckBox
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.ui.features.ProfileColor
+import ch.epfl.sdp.mobile.ui.features.social.ChessMatch
+import ch.epfl.sdp.mobile.ui.features.social.MatchResult
+import ch.epfl.sdp.mobile.ui.i18n.LocalLocalizedStrings
+
+val state =
+    object : ProfileState {
+      override val email: String = "badrtaddist1@gmail.com"
+      override val numberOfGames: Int = 10
+      override val numberOfPuzzles: Int = 10
+      override val matches: List<ChessMatch> =
+          List(20) { ChessMatch("Konor($it)", MatchResult.WIN, MatchResult.Reason.CHECKMATE, 27) }
+
+      override fun onSettingsClick() {}
+      override fun onEditClick() {}
+      override val backgroundColor: ProfileColor = ProfileColor.Pink
+      override val name: String = "Badr Taddist"
+      override val emoji: String = "😊"
+    }
+
+@Preview
+@Composable
+fun preview() {
+  //  SettingsButton(state::onSettingsClick)
+  // ProfileHeader(state = state)
+  //  GamesInfo(numPastGames = 42, numPuzzles = 10)
+  ProfileScreen(state = state)
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ProfileScreen(state: ProfileState, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+  LazyColumn(
+      verticalArrangement = Arrangement.Top,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = modifier) {
+    item { ProfileHeader(state) }
+    stickyHeader {
+      GamesInfo(numPastGames = state.numberOfGames, numPuzzles = state.numberOfPuzzles)
+    }
+    items(state.matches) { match ->
+      val title = strings.profileMatchTitle(match.adv)
+      val subtitle = strings.profileMatchInfo(match.matchResult, match.cause, match.numberOfMoves)
+      Match(title, subtitle, modifier = modifier)

```

Comment:
Bug : only the top-level composable within a `@Composable` function may use the `modifier` passed as a parameter.

```suggestion
      Match(title, subtitle)

```